dali/base/dasds.cpp

```diff
@@ -5390,9 +5398,18 @@ class CStoreHelper : implements IStoreHelper, public CInterface
         const char *editionBegin = name+strlen(base)+1;
         info.edition = atoi(editionBegin);
         if (iFileIO->size())
-            iFileIO->read(0, sizeof(unsigned), &info.crc);
+        {
+            iFileIO->read(0, sizeof(unsigned), &info.xmlCrc);
+            if (iFileIO->size())
```

Comment:
size() returns the size of the file, not the size remaining.  It is also worth reading all the data in one go - since the input is not buffered.

Simpler might be to structure the information read from the file into a nested structure and then have:

```
            // no check for size
            info.store.xmlCrc = 0;
            info.store.binaryCrc = 0;
            //Read as many fields from the store file as it contains..
            iFileIO->read(0, sizeof(info.store), &info.store);

```
Other implementations may be preferred

Reply:
>It is also worth reading all the data in one go - since the input is not buffered.

What Gavin was emphasising, was the efficiency of reading in 1 read call, vs multiple, and that can be done the way he suggests, i.e. put crc in a CrcInfo struct, and read all (read may read less than asked for)